mindmeld/app_manager.py

```diff
@@ -196,6 +211,12 @@ def parse(
         context = context or {}
 
         allowed_intents, nlp_params, dm_params = self._pre_nlp(params, verbose)
+
+        # if there is no language or locale in params we set them to app manager's
+        if not (nlp_params.get("language") or nlp_params.get("locale")):
```

Comment:
Since this code block is duplicated, make a helper function that can abstract this access pattern

Reply:
Done!